### PR TITLE
Feat : Implement sync with Meilisearch using node typescript

### PR DESCRIPTION
**Pull request by AvaterClasher**
## What does this PR do?

Implement a sync with meilisearch using node-typescript

## Test Plan

https://github.com/appwrite/templates/assets/116944847/eb65c958-8a49-45ae-bfd4-105f5dad2244

## Related PRs and Issues

#193 

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

Yes ✅

### CODE REVIEW

File: README.md
Comment (loks0n) [inline README.md:1]: Can you revert this file? It's generated automatically by our GH action :)

File: node-typescript/sync-with-meilisearch/README.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
+# ⚡ Sync Appwrite to Meilisearch Function
```
Comment (loks0n) [inline node-typescript/sync-with-meilisearch/README.md:1]: Can we change this to `Node.js (TypeScript) Sync Appwrite to Meilisearch Function`

File: node-typescript/sync-with-meilisearch/README.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+Sample `204` Response: No content.
+
+## ⚙️ Configuration
+
+| Setting           | Value         |
+| ----------------- | ------------- |
+| Runtime           | Node (18.0)   |
+| Entrypoint        | `src/main.ts` |
+| Build Commands    | `npm install` |
```
Comment (loks0n) [inline node-typescript/sync-with-meilisearch/README.md:25]: This needs to be `npm install && npm build`

File: node-typescript/sync-with-meilisearch/README.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+**Response**
+
+Sample `204` Response: No content.
+
+## ⚙️ Configuration
+
+| Setting           | Value         |
+| ----------------- | ------------- |
+| Runtime           | Node (18.0)   |
+| Entrypoint        | `src/main.ts` |
```
Comment (loks0n) [inline node-typescript/sync-with-meilisearch/README.md:24]: This needs to be `dist/main.js`

File: node-typescript/sync-with-meilisearch/src/utils.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
+import path from "path";
+import { fileURLToPath } from "url";
+import fs from "fs";
+
+export function throwIfMissing(obj: any, keys: string[]): void {
```
Comment (loks0n) [inline node-typescript/sync-with-meilisearch/src/utils.ts:5]: Can we add docstring descriptions to all the functions here:
e.g. for throwIfMissing
```typescript 
/**
* Throws an error if any of the keys are missing from the object
*/
```

**Review by loks0n — CHANGES_REQUESTED** [review]

File: node-typescript/sync-with-meilisearch/src/utils.ts (hunk 4 above)
Comment (AvaterClasher) [reply node-typescript/sync-with-meilisearch/src/utils.ts:5]: @loks0n Do i write the descriptions for all the main.ts functions too ?

Comment (loks0n) [reply node-typescript/sync-with-meilisearch/src/utils.ts:5]: Just the `utils.ts` file. 

We want to keep the template consistent between runtimes, and this is how the node javascript template looks :)

You can use the same descriptions as the node template also

**Comment by AvaterClasher** [comment]
@loks0n  Done with the changes

**Comment by loks0n** [comment]
Merged 🚀 

Thank you for you contribution!

**Comment by AvaterClasher** [comment]
@loks0n Can you give this the hactober-accepted label

**Review by loks0n — COMMENTED** [review]
LGTM

**Comment by gewenyu99** [comment]
Hey there! There were a lot of big PRs during this Hacktoberfest, and we wanted to give everyone ample time to collaborate with our engineering team. If you were able to merge your PRs during October, amazing. If it’s still not merged, don’t worry about it either. Either way, we’ve got your Hacktoberfest swag minted and ready to ship. 

Please comment with your Discord username here so we can contact you about your shipping information to deliver your Hacktoberfest swag.

**Comment by AvaterClasher** [comment]
Discord username - avaterclasher._07160

**Pull request merged**